exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/transactions/ThreadLocalTransactionManager.kt

```diff
@@ -249,36 +235,41 @@ fun <T> inTopLevelTransaction(
                 val answer = transaction.statement()
                 transaction.commit()
                 return answer
-            } catch (e: SQLException) {
-                handleSQLException(e, transaction, repetitions)
+            } catch (cause: SQLException) {
+                handleSQLException(cause, transaction, repetitions)
                 repetitions++
-                if (repetitions >= repetitionAttempts) {
-                    throw e
+                if (repetitions >= transaction.repetitionAttempts) {
+                    throw cause
+                }
+
+                if (retryInterval == null) {
+                    retryInterval = transaction.getRetryInterval()
+                    intermediateDelay = transaction.minRepetitionDelay
                 }
```

Comment:
These local variables need to be initialized once inside the `while` loop because the values they rely on are now being set inside the `statement` block instead of being provided as function parameters. So the `Transaction` properties are set to their defaults until `transaction.statement()` is called, which is why the computation for retries has to happen after the first try block.